### PR TITLE
fix(storefront): BCTHEME-345 define a main region on pages

### PR DESCRIPTION
**Pull request by bc-alexsaiannyi**
#### What?

This PR fixes an issue with not defining main region on pages by using HTML5 semantic tag `main`; for supporting this on IE11 `role='main' `has been [explicitly](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main) added  to ensure it is accessible in screen readers like JAWS. 

#### Tickets / Documentation

- [BCTHEME-345](https://jira.bigcommerce.com/browse/BCTHEME-345)

#### Screenshots (if appropriate)
main region on category page
main region on Homepage
main region on PDP

### CODE REVIEW

**Review by yurytut1993 — APPROVED** [review]

File: templates/components/common/body.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
-<div class="body" id='main-content' data-currency-code="{{currency_selector.active_currency_code}}">
+<main class="body" id='main-content' role='main' data-currency-code="{{currency_selector.active_currency_code}}">
```
Comment (BC-tymurbiedukhin) [inline templates/components/common/body.html:1]: Do we need role='main' if we are using main tag?

Comment (bc-alexsaiannyi) [reply templates/components/common/body.html:1]: it's recommended on MDN to add it for supporting old browsers like IE11

**Review by BC-tymurbiedukhin — APPROVED** [review]
LGTM

**Comment by mattcoy-arcticleaf** [comment]
@bc-alexsaiannyi  @BC-tymurbiedukhin @yurytut1993 , won't this cause issues with having 2 main elements on all the pages that already have a main element (page, contact-us, category, etc.)? https://html.spec.whatwg.org/multipage/grouping-content.html#the-main-element 
`A document must not have more than one main element that does not have the hidden attribute specified.`

On an unrelated note, is there a reason single quotes are used instead of double quotes?

/cc @bc-as @vasiliy-seleznev

**Comment by bc-alexsaiannyi** [comment]
@mattcoy-arcticleaf, you’re right. We’ve already created a [ticket](https://jira.bigcommerce.com/browse/BCTHEME-449) to address this issue. 

Regarding quotes - it was a tendency in writing without any specific reason behind and for consistency it will be overwritten in the same ticket. All changes will be delivered in the next release. Thanks for noting this to us.

**Comment by mattcoy-arcticleaf** [comment]
@bc-alexsaiannyi How would I log in to Jira?

**Comment by bc-alexsaiannyi** [comment]
@mattcoy-arcticleaf, you won't. it's for internal use.

**Pull request merged**